src/main/java/com/microsoft/jenkins/azuread/AzureSecurityRealm.java

```diff
@@ -926,8 +944,8 @@ public FormValidation doVerifyConfiguration(@QueryParameter final String clientI
         }
     }
 
-    private void updateIdentity(final AzureAdUser azureAdUser, final User u) {
-        if (azureAdUser != null && u != null) {
+    private void updateIdentity(final AzureAdUser azureAdUser, @NonNull final User u) {
+        if (azureAdUser != null) {
```

Comment:
as this can never actually be null in the 2 places its called I did a bit of refactoring to remove the null check and clean up a bit